stream/src/main/java/com/instaclustr/kafka/connect/stream/StreamSourceConnector.java

```diff
@@ -0,0 +1,174 @@
+package com.instaclustr.kafka.connect.stream;
+
+import com.instaclustr.kafka.connect.stream.codec.CodecError;
+import com.instaclustr.kafka.connect.stream.codec.Decoders;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.source.ExactlyOnceSupport;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.apache.kafka.connect.source.SourceConnectorContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toList;
+
+public class StreamSourceConnector extends SourceConnector {
+    private static final Logger log = LoggerFactory.getLogger(StreamSourceConnector.class);
+
+    public static final String DIRECTORY_CONFIG = "directory";
+    public static final String DIRECTORY_FILE_DISCOVERY_MINUTES = "directory.file.discovery.minutes";
+    public static final String FILES_CONFIG = "files";
+
+    public static final long DEFAULT_DIRECTORY_FILE_DISCOVERY_MINUTES = 60;
+
+    static final ConfigDef CONFIG_DEF = new ConfigDef()
+            .define(FILES_CONFIG, ConfigDef.Type.LIST, null, ConfigDef.Importance.HIGH, "Source filenames, default to files under the same directory.")
+            .define(DIRECTORY_CONFIG, ConfigDef.Type.STRING, null, ConfigDef.Importance.HIGH, "Source directory.")
+            .define(DIRECTORY_FILE_DISCOVERY_MINUTES, ConfigDef.Type.LONG, DEFAULT_DIRECTORY_FILE_DISCOVERY_MINUTES, ConfigDef.Importance.LOW, "The time to discover new directory files");
+
+    private AbstractConfig config;
+    
+    // Mutable state
+    private volatile List<String> files;
+    private Watcher directoryWatcher = null;
+
+    @Override
+    public void start(final Map<String, String> props) {
+        config = new AbstractConfig(CONFIG_DEF, props);
+        mustDefineDirectoryXorFiles(config);
+        if ((files = config.getList(FILES_CONFIG)) == null) {
+            Endpoint endpoint = Endpoints.of(props);
+            String directory = config.getString(DIRECTORY_CONFIG);
+            try {
+                files = endpoint.listRegularFiles(directory).collect(toList());
+            } catch (IOException e) {
+                throw new ConnectException(e);
+            }
+            var knownFiles = Set.copyOf(files);
+            directoryWatcher = Watcher.of(getDirectoryFileDiscoveryDuration());
+            directoryWatcher.watch(() -> {
+                var currentFiles = endpoint.listRegularFiles(directory).collect(toList());
```

Comment:
I am wondering if we should define a set of acceptable file types, in case there's invalid files uploaded to the bucket. But then we would need to specifically mention this in the documentation and may need to do further work in the future to add other file types.
I guess the responsibility of this should go to the customer - would be good to document this